### PR TITLE
Add a visual to show the what sections are run in what phases

### PR DESCRIPTION
**Pull request by simonsabin**
I felt a visual to show the stages executed in what phases would be useful.

### CODE REVIEW

**Comment by bravo-kernel** [comment]
I really like this :heart:

Looking at the preview, could you change the table header alignment, I think Github markdown supports use of  `:` for that.

image

File: docs/usage/data-driven-tests.mdx [hunk 1]
```diff
@@ -495,14 +495,28 @@ Describe "File - <_>" -ForEach $files {
...
+|`BeforeEach`| ✔️|❌ | ✔️
+|`It`| ✔️ |❌ | ✔️
+|`AfterEach`| ✔️|❌ | ✔️
+|`AfterAll`| ✔️|❌ | ✔️
+|`BeforeDiscovery`| N/A | ✔️ | ❌
+|`Describe` |✔️| ✔️ |❌
+|`Context` |✔️|✔️|❌
+| Other | ✔️|✔️|❌
+
+Other, is any script outside of `It`, `BeforeAll`, `BeforeEach`, `AfterEach` and `AfterAll`. You see that this code behaves like the new `BeforeDiscovery` block and thus for clarity on when script will be run its advised to use the `BeforeDiscovery`
```
Comment (bravo-kernel) [inline docs/usage/data-driven-tests.mdx:518]: Missing `.` at the end of the sentence.

File: docs/usage/data-driven-tests.mdx [hunk 2]
```diff
@@ -495,14 +495,28 @@ Describe "File - <_>" -ForEach $files {
 
 ## Migrating from Pester v4
 
-There are two main things to take into account when Data driven tests in Pester v5: The execution is not top down. And variables from `Discovery` are not available in `Run`, unless you explicitly attach them to a `Describe`, `Context` or `It`.
+There are two main things to take into account when Data driven tests in Pester v5: The [execution is not top down](#execution-is-not-top-down) like in V4 but done in two phases `Discovery` and `Run`. Variables from `Discovery` are not available in `Run`, unless you explicitly attach them to a `Describe`, `Context` or `It` using `Foreach` or `TestCases`
 
-This has big impact the classic pattern of using `foreach` to generate tests. 
+This has big impact the classic pattern of using `foreach` to generate tests,
```
Comment (bravo-kernel) [inline docs/usage/data-driven-tests.mdx:500]: I think the last char should be a `.`

**Comment by nohwnd** [comment]
LGTM, ready to merge?

**Review by nohwnd — APPROVED** [review]

**Comment by bravo-kernel** [comment]
Looks perfect to me, ship it!

Good job @simonsabin 👏

**Pull request merged**